ext/mysqli/mysqli_api.c

```diff
@@ -93,7 +93,7 @@ mysqli_escape_string_for_tx_name_in_comment(const char * const name)
 			{
 				*p_copy++ = v;
 			} else if (warned == FALSE) {
-				php_error_docref(NULL, E_WARNING, "Transaction name truncated. Must be only [0-9A-Za-z\\-_=]+");
+				php_error_docref(NULL, E_WARNING, "Transaction name has been truncated, since it can only contain the A-Z, a-z, 0-9, \"\\\", \"-\", \"_\", and \"=\" characters", ZSTR_VAL(function_name));
```

Comment:
Doesn't this still need an update to ext/mysqlnd/mysqlnd_connection.c?

Reply:
I don't understand how I didn't find the mysqlnd counterpart. Then please let me try to start this error promotion again.

Reply:
I don't think that mysqlnd is supposed to directly throw exceptions. It would have to go through the client error interface and this would end up being dependent on mysqli error handling mode.